### PR TITLE
Support the API of DELETE object

### PR DESCRIPTION
**Pull request by sfzeng**
**What this PR does / why we need it**:
This PR is to support the API of DELETE object.

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #
It fixed a part of [#691 ](https://github.com/opensds/multi-cloud/issues/691)

**Special notes for your reviewer**:

**Release note**:

```release-note
As the API of the DELETE object for YIG backend is not implemented yet, the deleting object which stored in a YIG backend is not tested.
```

### CODE REVIEW

File: s3/pkg/service/object.go [hunk 1]
```diff
@@ -262,12 +267,128 @@ func (s *s3Service) GetObject(ctx context.Context, in *pb.GetObjectInput, out *p
...
+		log.Errorln("failed to get backend with err:", err)
+		return err
+	}
+	sd, err := driver.CreateStorageDriver(backend.Type, backend)
+	if err != nil {
+		log.Errorln("failed to create storage, err:", err)
+		return err
+	}
+
+	err = sd.Delete(ctx, &pb.DeleteObjectInput{Bucket: bucket.Name, Key: objectKey, VersioId: obj.VersionId})
```
Comment (hopkings2008) [inline s3/pkg/service/object.go]: it is correct to delete meta first and then delete data in backend, or if we first delete data in backend successfully and then fail to delete meta, user can query object successfully but he cannot get that object.

File: s3/pkg/service/bucket.go [hunk 2]
```diff
@@ -134,49 +119,166 @@ func (s *s3Service) DeleteBucket(ctx context.Context, in *pb.Bucket, out *pb.Bas
...
 
 	bucket, err := s.MetaStorage.GetBucket(ctx, bucketName, false)
 	if err != nil {
 		log.Errorf("get bucket failed, err:%+v\n", err)
-		return err
+		return nil
 	}
 
 	// Check if bucket is empty
 	objs, _, _, _, _, err := s.MetaStorage.Db.ListObjects(ctx, bucketName, false, 1, nil)
```
[the remark this answers is not in the document]
Comment (sfzeng) [reply s3/pkg/service/bucket.go]: OK, it's done, thanks!

File: s3/pkg/service/object.go [hunk 3]
```diff
@@ -262,12 +267,128 @@ func (s *s3Service) GetObject(ctx context.Context, in *pb.GetObjectInput, out *p
...
+		// TODO: versioning
+		err = ErrInternalError
+	default:
+		log.Errorf("versioing of bucket[%s] is invalid:%s\n", bucket.Name, bucket.Versioning)
+		err = ErrInternalError
+	}
+
+	if err == nil {
+		// TODO: enable cache
+		/*
```
[the remark this answers is not in the document]
Comment (sfzeng) [reply s3/pkg/service/object.go]: Done, thanks!

File: s3/pkg/service/object.go (hunk 1 above)
Comment (sfzeng) [reply s3/pkg/service/object.go]: Thank you. I changed it to be like this: Mark the object as deleted in database, remove data in the backend, and then delete object from database.

File: s3/pkg/meta/db/drivers/tidb/bucket.go [hunk 4]
```diff
@@ -464,3 +467,41 @@ func (t *TidbClient) UpdateUsages(ctx context.Context, usages map[string]int64,
...
+
+		err = json.Unmarshal([]byte(lc), &tmp.LifecycleConfiguration)
+		if err != nil {
+			err = handleDBError(err)
+			return
+		}
+
+		buckets = append(buckets, &tmp)
+	}
+
```
[the remark this answers is not in the document]
Comment (sfzeng) [reply s3/pkg/meta/db/drivers/tidb/bucket.go]: Done! Thanks.

**Comment by hopkings2008** [comment]
LGTM

File: api/pkg/utils/obs/log.go [hunk 5]
```diff
@@ -69,13 +69,12 @@ func configureWriter(path, format string) {
 		TimestampFormat: defaultTimestampFormat,
 		LogFormat:       format + "\n",
 	})
-	fileWriter := &lumberjack.Logger{
+	multiWriter := io.MultiWriter(os.Stdout, &lumberjack.Logger{
```
Comment (kumarashit) [inline api/pkg/utils/obs/log.go]: good code to combine both

File: s3/pkg/meta/bucket.go [hunk 6]
```diff
@@ -49,35 +62,6 @@ func (m *Meta) GetBucket(ctx context.Context, bucketName string, willNeed bool)
...
 }
 
-/*
-func (m *Meta) GetBuckets() (buckets []*Bucket, err error) {
-	buckets, err = m.Client.GetBuckets()
-	return
-}
-*/
-
-func (m *Meta) UpdateUsage(ctx context.Context, bucketName string, size int64) error {
```
[the remark this answers is not in the document]
Comment (sfzeng) [reply s3/pkg/meta/bucket.go]: Hi @kumarashit  Usage is used for charging and the implementation will depends on cache(redis) which is not supported right now. We have an [issue](https://github.com/opensds/multi-cloud/issues/698) to trace it,  while added comment in s3/pkg/meta/object.go, line 98.

[the remark this answers is not in the document]
Comment (kumarashit) [reply s3/pkg/meta/bucket.go]: Ok

**Pull request merged**